src/lib/network_pool/network_pool_base.ml

```diff
@@ -1,16 +1,33 @@
 open Async_kernel
 open Core_kernel
 open Pipe_lib
+open Network_peer
 
 module Make (Transition_frontier : sig
   type t
-end)
-(Resource_pool : Intf.Resource_pool_intf
-                 with type transition_frontier := Transition_frontier.t) :
+end) (Resource_pool : sig
+  include Intf.Resource_pool_intf
+
+  (** Diff from a transition frontier extension that would update the resource pool*)
+  val handle_transition_frontier_diff :
+    transition_frontier_diff -> t -> unit Deferred.t
+
+  val create :
```

Comment:
These value defintions should probably just be added to `Intf.Resource_pool_intf` instead of being defined in the signature this functor takes.